### PR TITLE
Azure CLI Nat Gateway contract changes

### PR DESCRIPTION
**Pull request by khannarheams**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-natgateway/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (dumahesh-msft) [inline src/command_modules/azure-cli-natgateway/__init__.py]: Empty file?

File: src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/__init__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from azure.cli.core import AzCommandsLoader
+from azure.cli.command_modules.mymod._help import helps  # pylint: disable=unused-import
+
+class NatGatewayLoader(AzCommandsLoader):
+
+    def load_command_table(self, args):
+      from azure.cli.core.commands import CliCommandType
+
+      natgateway_custom = CliCommandType(
+        operations_tmpl='azure.cli.command_modules.maps#NatGatewayOperations.{}',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/__init__.py]: This should look like:
```Python
CliCommandType(operations_tmpl='azure.cli.command_modules.natgateway.custom#{}')
```

File: src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/__init__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      )
+
+      with self.command_group('network natgateway', natgateway_custom) as g:
```
Comment (tjprescott) [inline src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/__init__.py]: natgateway should be hyphenated `nat-gateway`.

Comment (tjprescott) [reply src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/__init__.py]: https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior

**Pull request merged**